Android/Narla Komal Kalyan/Advanced/NewsFeed/app/src/main/java/com/example/newsfeed/api/ApiClient.java

```diff
@@ -0,0 +1,69 @@
+package com.example.newsfeed.api;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import javax.security.cert.CertificateException;
+
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class ApiClient {
+
+    public static final String BASE_URL = "https://newsapi.org/v2/";
+    public static Retrofit retrofit;
+
+    public static Retrofit getApiClient() {
+
+        if(retrofit == null) {
+            retrofit = new Retrofit.Builder().baseUrl(BASE_URL)
+                    .client(getUnsafeOkHttpClient().build())
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .build();
+        }
+        return retrofit;
+    }
+
+    public static OkHttpClient.Builder getUnsafeOkHttpClient() {
+        try {
+            final TrustManager[] trustAllCerts = new TrustManager[] {
+                    new X509TrustManager() {
+                        @Override
+                        public void checkClientTrusted(java.security.cert.X509Certificate[] chain, String authType) {
```

Comment:
checkClientTrusted and checkServerTrusted are empty, which could cause insecure network traffic due to trusting arbitrary TLS/SSL certificates presented by peers. Always take care to handle these.

Reply:
Done